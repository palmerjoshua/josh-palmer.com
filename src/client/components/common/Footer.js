import React, { Component } from 'react';
import Splitter from './Splitter';

export default class Footer extends Component {
    static makeSmall(item) {
        return (
            <small>{item}</small>
        );
    }

    static makeFooterItem(item, useDelimiter) {
        let key = `footeritem${Math.random().toString(36).substring(7)}`;
        return (
            <span key={key}>
                {Footer.makeSmall(item)}
                {useDelimiter && <Splitter small={true}/>}
            </span>
        );
    }

    render() {
        let children = this.props.children;
        if (!Array.isArray(children))
            children = [children];
        return (
            <footer className="Footer" style={this.props.footerStyle}>
                    {children.map((child, i, arr) => (Footer.makeFooterItem(child, i < arr.length-1)))}
            </footer>
        );
    }
}
