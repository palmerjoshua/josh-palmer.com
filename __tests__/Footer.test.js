import React from 'react';
import Footer from '../src/client/components/common/Footer';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<Footer><p>first</p><p>second</p></Footer>).toJSON();
    expect(tree).toMatchSnapshot();
});
