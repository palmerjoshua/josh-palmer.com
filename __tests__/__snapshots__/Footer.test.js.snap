exports[`test renders correctly 1`] = `
<footer
  className="Footer"
  style={undefined}>
  <span>
    <small>
      <p>
        first
      </p>
    </small>
    <small
      className="Splitter">
       | 
    </small>
  </span>
  <span>
    <small>
      <p>
        second
      </p>
    </small>
  </span>
</footer>
`;
